rc/filetype/markdown.kak

```diff
@@ -31,6 +31,15 @@ hook -group markdown-highlight global WinSetOption filetype=markdown %{
     hook -once -always window WinSetOption filetype=.* %{ remove-highlighter window/markdown }
 }
 
+declare-option \
+    -docstring "map of supported languages in fenced code blocks. Keys determine which languages are recognised. The value names the corresponding highligher, if empty the key is used instead." \
+    str-to-str-map markdown_supported_languages \
+    awk= c= cabal= clojure= coffee= cpp= crystal= css= cucumber= d= diff= dockerfile= elixir= erlang= fish= \
+    gas= go= haml= haskell= hs=haskell html= ini= java= javascript= js=javascript json= julia= kak=kakrc kickstart= \
+    latex= lisp= lua= makefile= markdown= moon= objc= ocaml= perl= pug= python= ragel= \
+    ruby= rust= sass= scala= scss= sh= swift= toml= tupfile= typescript= yaml= sql=
```

Comment:
Here's an idea - since you mainly want to use this for

    set -add global markdown_supported_languages idris=idris

how about we do this automagically by assuming that a language identifier is the name of a filetype by default.


so when a markdown file with this contents is opened

	```foo
	```

we run `require-module foo` and give it a filetype=foo highlighter.
No user configuration needed.
This also means that we can remove the long list of filetypes from markdown.kak, which is currently a pain to keep in sync with supported languages anyway. 

Some more comments before I had that idea below